android/envoy/src/main/kotlin/org/greatfire/envoy/UrlUtil.kt

```diff
@@ -0,0 +1,111 @@
+package org.greatfire.envoy
+
+import android.util.Log
+import java.net.URI
+
+class UrlUtil {
+
+    companion object {
+
+        private val TAG = "UrlUtil"
+
+        fun sanitizeUrl(url: String, urlService: String): String {
```

Comment:
what do you think about moving to pull the service out from the url instead of making it a param?

Reply:
I added a "service unknown" constant that i use in places where i don't know the service.  i could add a convenience method that only takes a url and passed in that constant.  i think i want to retain the parameter because there are a few edge cases such as the local url for snowflake having an envoy prefix.

Reply:
so if the service is unknown, the method falls back on extracting it from the url?

Reply:
somewhat.  i look for socks5, https, or envoy, because those are the ones with duplicates.  otherwise i just get prefix and maybe port, which seems to cover the other cases which should be unique.